lcls-twincat-motion/Library/POUs/Motion/FB_EncSaveRestore.TcPOU

```diff
@@ -0,0 +1,59 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_EncSaveRestore" Id="{7f38772f-c9da-4f49-aee1-95cd4af3fbdf}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_EncSaveRestore
+VAR_IN_OUT
+    stMotionStage: DUT_MotionStage;
+END_VAR
+VAR_INPUT
+    bEnable: BOOL;
+END_VAR
+VAR_OUTPUT
+END_VAR
+VAR
+    fbSetPos: MC_SetPosition;
+    timer: TON;
+    tSaveDelay: TIME := T#10s;
+    bInit: BOOL;
+    bLoad: BOOL;
+    nLatchError: UDINT;
+END_VAR
+VAR PERSISTENT
+    bSaved: BOOL;
+    fPosition: LREAL;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[IF bEnable THEN
+    // Trigger a load if anything was saved at all
+    IF NOT bInit THEN
+        bInit := TRUE;
+        bLoad S= bSaved;
+        fbSetPos.Options.ClearPositionLag := TRUE;
+    END_IF
+
+    // Set our position if bLoad is true
+    fbSetPos(
+        Axis:=stMotionStage.Axis,
+        Execute:=bLoad,
+        Position:=fPosition);
+
+    // Only load once, at startup
+    bLoad R= fbSetPos.Done OR fbSetPos.Error;
+
+    // Keep the error latched, it can disappear if Execute is set to FALSE
+    IF fbSetPos.Error THEN
+        nLatchError := fbSetPos.ErrorID;
+    END_IF
+
+    // Do not save if we're currently loading
+    IF NOT bLoad THEN
+        fPosition := stMotionStage.stAxisStatus.fActPosition;
```

Comment:
I can check if there is an NC error, and I know the error range that corresponds to NC errors. That might be the best-effort try here to avoid saving a bad value.

Reply:
Addressed in https://github.com/pcdshub/lcls-twincat-motion/pull/108/commits/fca5cd0fa48999e252bb2a3af761c4ca283a3914